profiling/src/serializer/compressed_streaming_encoder.rs

```diff
@@ -17,6 +27,26 @@ impl CompressedProtobufSerializer {
         Ok(())
     }
 
+    /// Only meant for string table strings. This is essentially an
+    /// implementation of [prost::Message::encode] but for any `AsRef<str>`,
+    /// and specialized for handling the unlikely OOM conditions of writing
+    /// into a `Vec<u8>`.
+    pub(crate) fn encode_str(&mut self, item: impl AsRef<str>) -> anyhow::Result<()> {
+        let tag = 6u32;
```

Comment:
Instead of hardcoding the tag, could take it as a function argument and be general.

Reply:
The tag still has to be hard-coded somewhere. Previously it was here:

```rs
pub struct ProfileStringTableEntry {
    #[prost(string, required, tag = "6")]
    pub string_table_entry: String,
}
```

Would you rather I rename the method to something like `encode_string_table_entry` instead?

Reply:
I have renamed this method to `encode_string_table_entry`.